### PR TITLE
Actionpoint

### PR DESCRIPTION
**Pull request by katkelemen**

### CODE REVIEW

File: cards.py [hunk 1]
```diff
@@ -39,7 +40,8 @@ def attack(self, damage):
 
 class CreatureCard(Card):
     
-    
+    cost = 1
```
Comment (chiller) [inline cards.py:43]: I think spellcards and buildingcards should also have cost

File: game.py [hunk 2]
```diff
@@ -27,6 +27,13 @@ def add(self, cardclass, position):
                 self.player.discard_pile.append(self.buildings[position].__class__)
             self.buildings[position] = cardclass(self)
 
+    def add(self, cardclass, position):
+        if self.player.ap >= cardclass.cost:
+            self.player.ap -= cardclass.cost
+            self.add_without_cost(cardclass, position)
+
+
```
Comment (chiller) [inline game.py]: remove 2 new lines here

File: test.py [hunk 3]
```diff
@@ -96,6 +97,14 @@ def test_FieldOfNightmares(self):
...
         self.assertEqual(p2.hp, 15)
 
+    def test_action_points(self):
+        p1, p2 = self._game_factory([], [])
+        p1.ap, p2.ap = 2, 2
+        self.assertTrue(p1.ap)
+        self.assertEqual(p1.ap, 2)
+        p1.hand.append(CreatureCard)
+        p1.board.add(CreatureCard, 0)
+        self.assertEqual(p1.ap, 1)
```
Comment (chiller) [inline test.py:113]: you should also assert that you can't add more cards than ap

Comment (katkelemen) [inline test.py:113]: joo

**Pull request merged**